integration_test/upgrade_test.go

```diff
@@ -49,6 +50,31 @@ func TestKrewUpgrade(t *testing.T) {
 	}
 }
 
+func TestKrewUpgradeWhenPlatformNoLongerMatches(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithIndex().
+		Krew("install", validPlugin).
+		RunOrFail()
+
+	test.WithEnv("KREW_OS", "somethingelse")
+
+	// if upgrading 'all' plugins, must succeed
+	out := string(test.Krew("upgrade", "--no-update-index").RunOrFailOutput())
```

Comment:
:+1: for using the `--no-update-index` option. Should we add this to other integration tests to speed them up?

Reply:
Yeah not a bad idea, though worth seeing how much it speeds up. 
My motive here was to not to cause an index update b/c I was gonna modify the manifest file, but then i realized I can just override env.